### ISSUE TITLE
onEnter

### ISSUE THREAD

**Issue by bw1984** [jonathan-fielding/SimpleStateManager#45]
would it be possible to have the onEnter and onLeave events (or all events) return an object containing the details of the specific state and event? I rely on the console a lot for debugging and this would be very useful. For eaxmple;

```
ssm.getState('min_wide').attachCallback('enter', function(e)
{
	console.log(e);
}, true);
```

**Comment by jonathan-fielding** [comment]
Apologies, I only just saw this.

So `e` would be an object containing the details of the event that happened so something like:

{
    id: 'min_wide,
    method: 'enter',
    trigger: 'browser-resize'
}

Have I understood that right?

**Comment by bw1984** [comment]
Yes that would be great

**Comment by jonathan-fielding** [comment]
Ok I will look into it, are you using SSM 4?

**Comment by bw1984** [comment]
thanks. yes im using the latest version

**Comment by jonathan-fielding** [comment]
Awesome

**Comment by jonathan-fielding** [comment]
@bw1984 I have this screenshot to show the progress, does this solve your problem

<img width="747" alt="screen shot 2018-03-06 at 22 29 53" src="https://user-images.githubusercontent.com/630034/37062337-09586ee6-218e-11e8-948e-ef3b2c0b988e.png">

**Comment by jonathan-fielding** [comment]
https://github.com/jonathan-fielding/SimpleStateManager/pull/47 Pull request I made with this workign implementation,

**Comment by jonathan-fielding** [comment]
Have released 4.1.0 with this change https://www.npmjs.com/package/simplestatemanager
